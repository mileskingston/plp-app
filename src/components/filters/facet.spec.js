import React from 'react';
import { shallow } from 'enzyme';
import Facet from './facet';

describe('Facet Component', () => {
  const props = {
    facet: {
      value: 'value 1',
      name: 'name 1',
      entry: [{
        count: 2,
        label: 'entry label 1',
        value: 'entry 1',
      }, {
        count: 1,
        label: 'entry label 2',
        value: 'entry 2',
      }],
    },
    onClick: jest.fn(),
  };
  it('renders component', () => {
    expect(shallow(<Facet {...props} />)).toMatchSnapshot();
  });

  it('renders component', () => {
    let component = shallow(<Facet {...props} />);
    component.find('.facet-entry__input').first().simulate('click');
    expect(props.onClick).toHaveBeenCalled();
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});