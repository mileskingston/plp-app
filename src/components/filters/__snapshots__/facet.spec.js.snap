// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facet Component renders component 1`] = `
<div
  className="facet"
  key="value 1"
>
  <h3>
    name 1
  </h3>
  <label
    className="facet-entry"
    htmlFor="entry label 1"
    key="0"
  >
    <input
      className="facet-entry__input"
      id="entry label 1"
      name="entry label 1"
      onClick={[Function]}
      tabIndex="0"
      type="checkbox"
      value="entry 1"
    />
    <span
      className="facet-entry__label"
    >
      entry label 1
       
      <span>
        2
      </span>
    </span>
  </label>
  <label
    className="facet-entry"
    htmlFor="entry label 2"
    key="1"
  >
    <input
      className="facet-entry__input"
      id="entry label 2"
      name="entry label 2"
      onClick={[Function]}
      tabIndex="0"
      type="checkbox"
      value="entry 2"
    />
    <span
      className="facet-entry__label"
    >
      entry label 2
       
      <span>
        1
      </span>
    </span>
  </label>
</div>
`;
